homeassistant/components/recorder/const.py

```diff
@@ -42,6 +42,15 @@
 # capped at 4000 to avoid performance issues
 SQLITE_MODERN_MAX_BIND_VARS = 4000
 
+# As soon as we have more than 999 ids, split the query as the
+# MariaDB/MySQL optimizers handle it poorly and will no longer
+# do an index only scan
+# https://github.com/home-assistant/core/issues/132865
+# We do this for all engines to keep the code simple and
+# ensure we do not hit any future limits since all engines
+# have a limit on the number of bind vars
+MAX_IDS_FOR_START_TIME_QUERY = 999
```

Comment:
Did you find some reference in MySQL / MariaDB documentation or source code which we could add a reference too in addition to the core issue?

Reply:
There is nothing documented about in the mariadb or MySQL docs, or anything useful in the code to link to in order to point out.  I think this is such a corner case. They didn't really even consider it when I dug to the Maria DB bug tracker, the prevailing advice to previous issue reports from them was to open a feature request to force an index scan.